### PR TITLE
Document using `next` to stop executing a `lane`

### PR DESCRIPTION
**Pull request by mfurtak**
Documenting the research findings for #5913

### CODE REVIEW

File: fastlane/docs/Advanced.md [hunk 1]
```diff
@@ -90,6 +90,44 @@ lane :calculate do |options|
...
+  puts "As well as this!"
+end
+
+private_lane :second_lane do |options|
+  next
+  puts "This won't be shown"
+end
+```
+
+When you stop executing a lane early with `next`, any `after_each` and `after_all` blocks you have will still trigger as usual :+1:
```
Comment (KrauseFx) [inline fastlane/docs/Advanced.md]: 👍

**Review by KrauseFx — APPROVED** [review]
This is great 👍  Nice writeup

**Comment by KrauseFx** [comment]
I just noticed, this should also be updated https://github.com/fastlane/docs, where we also have an `Advanced.md`. Currently those files are not synced automatically, as we probably want to remove all remaining docs from the main repo in the future

**Comment by KrauseFx** [comment]
@mfurtak can you also copy that to docs.fastlane.tools or should I do it?

**Comment by mfurtak** [comment]
Yes, I'll edit it there as well

**Pull request merged**